rest-client/src/test/java/org/sagebionetworks/bridge/rest/ConfigTest.java

```diff
@@ -24,15 +25,100 @@ public void test() {
     }
     
     @Test
-    @Ignore // will not change environment on Travis
+    public void read_fromEnv() {
+        Config config = spy(Config.class);
+        when(config.readEnv("FOO_BAR")).thenReturn("baz");
+        
+        String retValue = config.read("foo.bar", null);
+        assertEquals("baz", retValue);
+    }
+    
+    @Test
+    public void read_fromEnvWithPrefix() {
+        Config config = spy(Config.class);
+        config.set(Environment.LOCAL);
+        
+        when(config.readEnv("FOO_BAR")).thenReturn("baz");
+        when(config.readEnv("LOCAL_FOO_BAR")).thenReturn("biff");
+        
+        String retValue = config.read("FOO_BAR", null);
+        assertEquals("biff", retValue);
+    }
+    
+    @Test
+    public void read_fromSystemProperty() {
+        Config config = spy(Config.class);
+        when(config.readSystemProp("foo.bar")).thenReturn("baz");
+        
+        String retValue = config.read("FOO_BAR", null);
+        assertEquals("baz", retValue);
+    }
+    
+    @Test
+    public void read_fromSystemPropertyWithPrefix() {
+        Config config = spy(Config.class);
+        config.set(Environment.LOCAL);
+        
+        when(config.readSystemProp("foo.bar")).thenReturn("baz");
+        when(config.readSystemProp("local.foo.bar")).thenReturn("biff");
+        
+        String retValue = config.read("FOO_BAR", null);
+        assertEquals("biff", retValue);
+    }
+    
+    @Test
+    public void read_envOverridesSystemProperty() {
+        // We can't spy on the read methods before we construct config and at that time,
+        // all the properties are read in. We *can* test environmental overrides in the 
+        // properties file and we can test overrides of the environment by system properties.
+        Config config = spy(Config.class);
+        when(config.readEnv("FOO")).thenReturn("env testName");
+        when(config.readSystemProp("foo")).thenReturn("system prop testName");
+        
+        String retValue = config.read("foo", null);
+        assertEquals("env testName", retValue);
+    }
```

Comment:
Manually test override of properties file.

Reply:
Verified locally that even when the properties file are present and correct to run the tests locally, the Jenkins system property overrides on the command line will cause the tests to run correct against another environment.